app/components/markdown/markdown_link/markdown_link.js

```diff
@@ -12,36 +12,54 @@ import mattermostManaged from 'app/mattermost_managed';
 
 import Config from 'assets/config';
 
+import {escapeRegex} from 'app/utils/markdown';
+import {preventDoubleTap} from 'app/utils/tap';
 import {normalizeProtocol} from 'app/utils/url';
 
 export default class MarkdownLink extends PureComponent {
     static propTypes = {
         children: CustomPropTypes.Children.isRequired,
         href: PropTypes.string.isRequired,
         onLongPress: PropTypes.func,
+        onPermalinkPress: PropTypes.func,
+        serverURL: PropTypes.string.isRequired,
+        siteURL: PropTypes.string.isRequired,
     };
 
     static defaultProps = {
         onLongPress: () => true,
+        onPermalinkPress: () => true,
     };
 
     static contextTypes = {
         intl: intlShape.isRequired,
     };
 
-    handlePress = () => {
-        const url = normalizeProtocol(this.props.href);
+    handlePress = preventDoubleTap(() => {
+        const {href, onPermalinkPress, serverURL, siteURL} = this.props;
+        const url = normalizeProtocol(href);
 
         if (!url) {
             return;
         }
 
-        Linking.canOpenURL(url).then((supported) => {
-            if (supported) {
-                Linking.openURL(url);
-            }
-        });
-    };
+        const pattern = new RegExp('^' + escapeRegex(serverURL) + '\\/([^\\/]+)\\/pl\\/(\\w+)');
+        const sitePattern = new RegExp('^' + escapeRegex(siteURL) + '\\/([^\\/]+)\\/pl\\/(\\w+)');
+        const match = url.match(pattern) || url.match(sitePattern);
+
+        if (!match) {
+            Linking.canOpenURL(url).then((supported) => {
+                if (supported) {
+                    Linking.openURL(url);
+                }
+            });
+            return;
+        }
+
+        const teamName = match[1];
+        const postId = match[2];
+        onPermalinkPress(postId, teamName);
```

Comment:
I don't think https://github.com/mattermost/mattermost-mobile/pull/1449#discussion_r169663880 was addressed here

Reply:
then I don't follow, the team name is passed down to the onPermalinkPress prop and then we load the channels for that team, what am I missing?

Reply:
I mean that we could just pass down the permalink and have the post component (or whatever defines onPermalinkPress) call it. I'm 0/5 though now since I didn't realize that this is the only place that we were parsing the permalink.